### DISCUSSION TITLE
Error running with Docker: Could not find gem 'jekyll-diagrams' in locally installed gems. (Bundler::GemNotFound)

### DISCUSSION THREAD

**Discussion by pdeperth** [alshedivat/al-folio#2042] [Q&A] [answered]
Hello! 

Do any of you encounter this error when running the website with docker?
After a successful `docker compose pull` and then `docker compose up`, I get the error: 
`/var/lib/gems/3.0.0/gems/bundler-2.5.3/lib/bundler/resolver.rb:332:in 'raise_not_found!': Could not find gem 'jekyll-diagrams' in locally installed gems. (Bundler::GemNotFound)`
I ran `gem install jekyll-diagrams` and restarted the docker service but I still get the same error afterwards.
Thank you for your help.

**Comment by george-gca** [comment]
Update your repository to the latest version. #1992 changed the way mermaid diagrams are supported in the template.

**Reply by pdeperth** [reply]
Dear George, 
thank you for your answer. 
I have rebased my repository on the latest `git@github.com:alshedivat/al-folio.git`, but I still have the same error unfortunately.
Best.

**Reply by pdeperth** [reply, marked as answer]
Dear George,
it is okay, now I have solved the issue: it was coming from a permission problem preventing the `bundle install` command from running successfully. 
Best.

**Reply by shashikantilager** [reply]
Hi, how did you solve permission problem?

**Reply by george-gca** [reply]
`jekyll-diagrams` lib is not installed in the current docker image. Actually it hasn't been installed for a while, because we changed how it is used on the template. That's why it is failing for your case. You can either create your own image, use an older one, or update your code to the latest version.

**Reply by PierreExeter** [reply]
I have a similar issue. I'm trying to install the gem `'jekyll-redirect-from'`. Here are the steps I took :
1. I added `gem "jekyll-redirect-from"` to the Gemfile
2. I added ` jekyll-redirect-from` in the plugin list of the `_config.yml` file
3. I ran `docker compose pull`
4. I ran `docker compose up`

I get the following error : `/usr/local/bundle/gems/bundler-2.5.18/lib/bundler/resolver.rb:354:in 'raise_not_found!': Could not find gem 'jekyll-redirect-from' in locally installed gems. (Bundler::GemNotFound)`

I guess this is because the `bundle install` command is not executed properly. But I don't know how to fix it. 

How can I install additional gems in the docker image ? 
(sorry I'm quite new to Docker)

**Reply by PierreExeter** [reply]
Ok I found the solution ! I had to build a new Docker image with :
`$ docker compose up --build`

**Reply by anteromontonio** [reply]
> Ok I found the solution ! I had to build a new Docker image with : `$ docker compose up --build`

I tried this (in order to add the gem j[ekyll-fetch-notion](https://github.com/seroperson/jekyll-fetch-notion) and id didn't work
